tests/Operation/WatchFunctionalTest.php

```diff
@@ -924,17 +932,22 @@ public function testRewindExtractsResumeTokenAndNextResumes()
         $changeStream->rewind();
         $this->assertFalse($changeStream->valid());
 
-        $changeStream->next();
-        $this->assertTrue($changeStream->valid());
+        $this->advanceCursorUntilValid($changeStream);
 
         $resumeToken = $changeStream->current()->_id;
         $options = ['resumeAfter' => $resumeToken] + $this->defaultOptions;
         $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $options);
         $changeStream = $operation->execute($this->getPrimaryServer());
-        $this->assertSame($resumeToken, $changeStream->getResumeToken());
+        $this->assertEquals($resumeToken, $changeStream->getResumeToken());
 
         $changeStream->rewind();
-        $this->assertTrue($changeStream->valid());
+
+        if ($this->isShardedCluster()) {
+            $this->advanceCursorUntilValid($changeStream);
```

Comment:
Even though `rewind()` should never execute a `getMore` after [PHPLIB-451](https://jira.mongodb.org/browse/PHPLIB-451), the `aggregate` command on the sharded cluster is not guaranteed to return any initial results for the same reason that `getMore` may not (due to waiting for a periodic noop oplog entry). Is that correct? If so, I think a comment might be prudent here (and in the two instances below), since this is a bit more complicated than other places where you were able to simply replace `next()` and a valid assertion with `advanceCursorUntilValid()` directly.